tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -89,6 +89,17 @@ sub can_invite_unbound_3pid
    })->then( sub {
       $id_server->bind_identity( $hs_uribase, "email", $invitee_email, $invitee );
    })->then( sub {
+      matrix_get_room_state( $inviter, $room_id,
+         type      => "m.room.member",
+         state_key => $invitee->user_id,
+      )
+   })->then( sub {
+      my ( $body ) = @_;
+      log_if_fail "Body", $body;
+      $body->{third_party_invite}{display_name} eq "Bob" or
+         die "Expected third_party_invite display_name to be 'Bob'";
+      Future->done( 1 );
+   })->then( sub {
```

Comment:
The entire construction of

       Future->done(1);
    })->then( sub {

is effectively redundant. You can safely delete these two lines; though I'd suggest replacing them with an empty one to keep the separation between the two logical phases.